pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -355,3 +355,34 @@ func DeleteURLsConfigMap(revised *corev1.ConfigMap, crName string, newURLs []cor
 		}
 	}
 }
+
+// changeMonitoringConsoleAnnotations updates the checkUpdateImage field of the Monitoring Console Annotations to trigger the reconcile loop
+// on update, and returns error if something is wrong.
+func changeMonitoringConsoleAnnotations(ctx context.Context, client splcommon.ControllerClient, cr *enterpriseApi.ClusterManager) error {
```

Comment:
Nit: Is there a way where we can make this API generic? I know there are some places where the API has to be CR specific.
cc: @vivekr-splunk

Reply:
Also, we can have an API like `changeAnnotations` as a generic API and depending on the type of CR take a specific action i.e the dependencies might be different based on the CRs.

Reply:
There is already a `changeAnnotations` API which does the common actions for all the instanceTypes. To have a fully generic API, the number of arguments which will be required will increase because we want to get the instance from the reference along with the image of the caller instance which I think can become a bit cluttered.
Do you reckon the common API will be a better way to go forward?

Reply:
Makes sense to keep this as-is.